src/debugAdapter/goDebug.ts

```diff
@@ -431,12 +431,11 @@ class Delve {
 						if (launchArgs.buildFlags) {
 							runArgs.push(launchArgs.buildFlags);
 						}
+						runArgs.push(program);
```

Comment:
from src/goDebugConfiguration.ts, the program can be set to the active editor's document file name. If the program is set to the file name, I'm afraid it's basically reverting the intended fix for #3096 (supporting the main package with multiple files). What am I missing?

Reply:
my mistake, it turely reverts the intended fix for #3096 , but i found a win-win solution, if the program is set to `*.go` instead of `.`, the executable filename will be named as the filename of main function, and it can support multiple go files in a folder.

Reply:
Thank you for addressing the comment. In that case, the executable filename will be the first file name. 

Naming the binary based on the directory name is the de-facto standard of the go command and users who'd get the command using `go get` command will depend on the behavior. For example, golang.org/x/tools/gopls has main.go as the only file but we expect the final binary name to be the directory name 'gopls', not 'main'.  Why is it a problem?

Reply:
i agree with your opinion, actually, i thought the executable name should be same in both debug and run mode. but it's not a standard.